core/integration/features/orders/limit-orders-all-market-types.feature

```diff
@@ -0,0 +1,143 @@
+      | party1| ETH/DEC23 | buy  | 1      | 900   | 0                | TYPE_LIMIT | TIF_IOC |         |       |
+      | party1| ETH/DEC23 | buy  | 1      | 900   | 0                | TYPE_LIMIT | TIF_IOC | post    |       |
+      | party1| ETH/DEC23 | buy  | 1      | 900   | 0                | TYPE_LIMIT | TIF_IOC | reduce  | OrderError: reduce only order would not reduce position |
+      | party1| ETH/DEC23 | sell | 1      | 1100  | 0                | TYPE_LIMIT | TIF_IOC |         |       |
+      | party1| ETH/DEC23 | sell | 1      | 1100  | 0                | TYPE_LIMIT | TIF_IOC | post    |       |
+      | party1| ETH/DEC23 | sell | 1      | 1100  | 0                | TYPE_LIMIT | TIF_IOC | reduce  | OrderError: reduce only order would not reduce position |
+
+    # A matching IOC should be accepted
+    When the parties place the following orders:
+      | party | market id | side | volume | price | resulting trades | type       | tif     | only    | error |
+      | party1| ETH/DEC23 | buy  | 1      | 1100  | 1                | TYPE_LIMIT | TIF_IOC |         |       |
+      | party1| ETH/DEC23 | buy  | 1      | 1100  | 0                | TYPE_LIMIT | TIF_IOC | post    | OrderError: post only order would trade |
+      | party1| ETH/DEC23 | buy  | 1      | 1100  | 0                | TYPE_LIMIT | TIF_IOC | reduce  | OrderError: reduce only order would not reduce position |
+      | party1| ETH/DEC23 | sell | 1      | 900   | 1                | TYPE_LIMIT | TIF_IOC |         |       |
+      | party1| ETH/DEC23 | sell | 1      | 900   | 0                | TYPE_LIMIT | TIF_IOC | post    | OrderError: post only order would trade |
+      | party1| ETH/DEC23 | sell | 1      | 900   | 0                | TYPE_LIMIT | TIF_IOC | reduce  | OrderError: reduce only order would not reduce position |
+
+    # A non matching FOK should be accepted
+    When the parties place the following orders:
+      | party | market id | side | volume | price | resulting trades | type       | tif     | only    | error |
+      | party1| ETH/DEC23 | buy  | 1      | 900   | 0                | TYPE_LIMIT | TIF_FOK |         |       |
+      | party1| ETH/DEC23 | buy  | 1      | 900   | 0                | TYPE_LIMIT | TIF_FOK | post    |       |
+      | party1| ETH/DEC23 | buy  | 1      | 900   | 0                | TYPE_LIMIT | TIF_FOK | reduce  | OrderError: reduce only order would not reduce position |
+      | party1| ETH/DEC23 | sell | 1      | 1100  | 0                | TYPE_LIMIT | TIF_FOK |         |       |
+      | party1| ETH/DEC23 | sell | 1      | 1100  | 0                | TYPE_LIMIT | TIF_FOK | post    |       |
+      | party1| ETH/DEC23 | sell | 1      | 1100  | 0                | TYPE_LIMIT | TIF_FOK | reduce  | OrderError: reduce only order would not reduce position |
+
+    # A matching FOK should be accepted
+    When the parties place the following orders:
```

Comment:
It shouldn't matter, but might be useful to copy the same scenario but place orders with ticks to see if some behaviour changes if we add the variable of time